added option to purchase premium plus

matthias-luger opened the pull request.

matthias-luger committed:
added option to purchase premium plus

matthias-luger commented:
TODO: add correct premium+ productId

matthias-luger committed:
fix refactored constant

matthias-luger committed:
- get current premium type by priority
- use new endpoint to get premium products
- removed old endpoint

matthias-luger committed:
add information if a different premium product that active is bought

matthias-luger committed:
- add starter premium
- fix up styling and duration options for starter premium
- fix up wordings
- add logic to find premium products from changed slugs

matthias-luger committed:
fix premium type lookup

matthias-luger committed:
- fixed error responses to be shown
- fixed wording in purchase success message
- fixed itemDetails page crash for non existing items
- fixed bug that caused coflcoins balance call before login

matthias-luger committed:
- parse new format of premium products correctly
- fix bug in the api error handling

matthias-luger committed:
fix new format of premium products

matthias-luger committed:
- removed useless payment component
- added new premium products to the premium feature list
- reload active premium product after purchase

Ekwav commented on api/ApiHelper.tsx:
```diff
@@ -1620,6 +1604,38 @@ export function initAPI(returnSSRResponse: boolean = false): API {
+        return new Promise((resolve, reject) => {
+            let googleId = localStorage.getItem('googleId')
+            if (!googleId) {
+                toast.error('You need to be logged in to save privacy settings.')
```
Please don't copy paste code, either refactor or delete it completely

matthias-luger replied:
I dont see a good way to refactor this...
The `if` and `return` cant be extracted and I need the googleId a few lines below anyways.

The only thing wrong here is that the message should be a different. I fixed that

Ekwav commented on api/ApiHelper.tsx:
```diff
@@ -518,23 +519,6 @@ export function initAPI(returnSSRResponse: boolean = false): API {
         })
     }
 
-    let hasPremium = (googleId: string): Promise<Date> => {
```
![image](https://user-images.githubusercontent.com/16632490/186516054-24f85086-560e-4fa6-92e1-706ee4e6649b.png)

matthias-luger committed:
fix premium product error message

matthias-luger commented:
@Ekwav  I changed the text of the title card on the flipper page, as this is not longer true. Please take a look:
![image](https://user-images.githubusercontent.com/58751503/186558004-449eb040-58e3-4368-9756-9bf2a757c192.png)

matthias-luger committed:
- refactored old premium checks to new products endpoint
- changed title of flipper page according to different premium products

matthias-luger committed:
Merge branch 'develop' into premium-plus

Ekwav commented on components/Premium/PremiumFeatures/PremiumFeatures.tsx:
```diff
@@ -22,62 +25,114 @@ function PremiumFeatures() {
-                        <td>{xIcon}</td>
-                        <td>{checkIcon}</td>
+                        <td className={styles.featureColumn}>Kat flips</td>
+                        <td className={styles.premiumProductColumn}>{xIcon}</td>
```
wouldn't it make sense to make the icon variable include the td element?

Ekwav commented on components/Premium/PremiumFeatures/PremiumFeatures.tsx:
```diff
@@ -22,62 +25,114 @@ function PremiumFeatures() {
                     </tr>
                 </thead>
                 <tbody>
                     <tr>
```
please add chat colors (gray, white, green, gold)   
and support the project (❌,✅,✅,✅)  
bazaar data history (✅,✅,✅,✅)

Ekwav commented on components/Premium/PremiumFeatures/PremiumFeatures.tsx:
```diff
@@ -22,62 +25,114 @@ function PremiumFeatures() {
-                        <td className={styles.featureColumn}>Priority feature request</td>
-                        <td>{xIcon}</td>
-                        <td>{checkIcon}</td>
+                        <td className={styles.featureColumn}>Up to 100 notifications</td>
+                        <td className={styles.premiumProductColumn}>{xIcon}</td>
+                        <td className={styles.premiumProductColumn}>{checkIcon}</td>
+                        <td className={styles.premiumProductColumn}>{checkIcon}</td>
+                        <td className={styles.premiumProductColumn}>{checkIcon}</td>
```
please change notifications to numbers (3,10,100,1000)

Ekwav commented on components/Premium/PremiumFeatures/PremiumFeatures.tsx:
```diff
@@ -22,62 +25,114 @@ function PremiumFeatures() {
+                        <td className={styles.premiumProductColumn}>{checkIcon}</td>
                     </tr>
                     <tr>
                         <td className={styles.featureColumn}>Discord role "Flipper" (on request)</td>
```
discord roles (none,none,flipper,sniper)

Ekwav commented on components/Premium/PremiumFeatures/PremiumFeatures.tsx:
```diff
@@ -22,62 +25,114 @@ function PremiumFeatures() {
-                        <td className={styles.featureColumn}>Full access to flipper filters</td>
-                        <td>{xIcon}</td>
-                        <td>{checkIcon}</td>
+                        <td className={styles.featureColumn}>Craft flips</td>
```
```suggestion
                        <td className={styles.featureColumn}>Top 3 Craft flips</td>
```

Ekwav commented on components/Premium/PremiumFeatures/PremiumFeatures.tsx:
```diff
@@ -22,62 +25,114 @@ function PremiumFeatures() {
-                        <td className={styles.featureColumn}>Faster Flipper without delay</td>
-                        <td>{xIcon}</td>
-                        <td>{checkIcon}</td>
+                        <td className={styles.featureColumn}>Kat flips</td>
```
```suggestion
                        <td className={styles.featureColumn}>Top 3 Kat flips</td>
```

Ekwav commented on components/Flipper/Flipper.tsx:
```diff
@@ -407,6 +407,48 @@ function Flipper(props: Props) {
+            if (type.priority === PREMIUM_RANK.PREMIUM_PLUS) {
+                return (
+                    <span>
+                        You are using premium+. Check details <Link href="/premium">here</Link>
```
please make an element to display the accountTier and use it everywhere. on the account page it displays it incorrectly for example 
![image](https://user-images.githubusercontent.com/16632490/187764961-a3a966ca-986c-4e77-9c3d-396e8a2d9f30.png)

Ekwav commented on components/Flipper/Flipper.tsx:
```diff
@@ -407,6 +407,48 @@ function Flipper(props: Props) {
+            if (type.priority === PREMIUM_RANK.STARTER) {
+                return (
+                    <span>
+                        You are using starter premium. Check details <Link href="/premium">here</Link>
```
why not make `starter premium` clickable instead of writing check details here?

Ekwav commented on components/Premium/BuyPremium/BuyPremium.tsx:
```diff
@@ -50,11 +82,29 @@ function BuyPremium() {
+                <ul>
+                    <li>
+                        <span className={styles.label}>Type:</span>
+                        {purchasePremiumType.label}
```
could the tier select be made into a toggle-button like this: https://material.angular.io/components/button-toggle/overview
and then show the extend time options in a dropdown.  
For starter I would suggest dropdown options of 
* 1 day
* 1 week
* 4 weeks
* 6 months

matthias-luger committed:
Update components/Premium/PremiumFeatures/PremiumFeatures.tsx

Co-authored-by: Äkwav <16632490+Ekwav@users.noreply.github.com>

matthias-luger committed:
Update components/Premium/PremiumFeatures/PremiumFeatures.tsx

Co-authored-by: Äkwav <16632490+Ekwav@users.noreply.github.com>

matthias-luger committed:
- update premium features
- update flipper premium link

matthias-luger committed:
Merge branch 'premium-plus' of https://github.com/Coflnet/hypixel-react into premium-plus

matthias-luger committed:
- changed premium type selection to toggle buttons
- changed selection to only select one starter premium
- fixed premium status on account page

matthias-luger committed:
Merge branch 'develop' into premium-plus

matthias-luger committed:
change sniper color

Ekwav commented on components/Premium/PremiumFeatures/PremiumFeatures.tsx:
```diff
@@ -22,62 +29,143 @@ function PremiumFeatures() {
+                        <td className={styles.premiumProductColumn}>-</td>
+                        <td className={styles.premiumProductColumn}>-</td>
+                        <td className={styles.premiumProductColumn}>Flipper</td>
+                        <td className={styles.premiumProductColumn}>Sniper</td>
```
Add a note stating that the role will be given out in the second week of September, the bot currently can't do it

Ekwav replied:
or just make it say flipper for now

Ekwav commented on components/Premium/PremiumFeatures/PremiumFeatures.tsx:
```diff
@@ -22,62 +29,143 @@ function PremiumFeatures() {
+                                type="hover"
+                                tooltipContent={
+                                    <p>
+                                        Hypixel updates once every 60 seconds. After the we receive the new auctions this is how long it will take for you to
+                                        get the flips.
```
```suggestion
                                        The Hypixel Auctions API updates once every 60 seconds. 
                                        After we were able to load new auctions, this is how long it will take until they are shown to you. (Parsing auctions, finding references, comparing to determine profit, distributing and filtering and seinding to you) 
```

Ekwav commented on utils/PremiumTypeUtils.tsx:
```diff
@@ -0,0 +1,78 @@
+        options: [
+            { value: 1, label: '1 day', productId: 'starter_premium-day', price: 24 },
+            { value: 1, label: '1 week', productId: 'starter_premium-week', price: 120 },
+            { value: 4, label: '4 weeks', productId: 'starter_premium-week', price: 1201 },
```
```suggestion
            { value: 4, label: '4 weeks', productId: 'starter_premium-week', price: 120 },
```

Ekwav commented on components/Premium/PremiumFeatures/PremiumFeatures.tsx:
```diff
@@ -22,62 +29,143 @@ function PremiumFeatures() {
+                            List of low supply items (<Link href="/lowSupply">here</Link>)
+                        </td>
+                        {xIconElement}
+                        {checkIconElement}
```
don't include this in starter for now
```suggestion
                        {xIconElement}
```

matthias-luger committed:
fix typo

matthias-luger committed:
Update components/Premium/PremiumFeatures/PremiumFeatures.tsx

Co-authored-by: Äkwav <16632490+Ekwav@users.noreply.github.com>

matthias-luger committed:
dont show lowSupplyItems for starter premium

matthias-luger committed:
Merge branch 'premium-plus' of https://github.com/Coflnet/hypixel-react into premium-plus

matthias-luger committed:
dont include low supply items in starter premium

matthias-luger committed:
add coming soon note to Sniper rank

matthias-luger committed:
Change premium+ discord role to flipper
